sdlib/d/gc/region.d

```diff
@@ -380,12 +404,12 @@ struct Region {
 		assert(dirtyBlockCount <= blockCount,
 		       "Dirty block count exceeds block count!");
 
-		address = ptr;
+		bits = (cast(ulong) ptr) | generation & GenerationMask
+			| ulong(getFreeSpaceClass(blockCount) & AllocClassMask)
+				<< AllocClassIndex;
```

Comment:
bits = ...
bits |= ...
bits |= ...

Also, is there a reason we are masking ubytes? Either they can be out of bound, but then we should probably assert they are int he range we expect, or they are not. in this case, they are not.